### PR TITLE
Add basic docs for linters

### PR DESCRIPTION
**Pull request by maetl**
I'm really impressed by some of the ideas in ROM and it has been a lot of fun to play with writing extensions so far. While rapidly growing and stabilising, ROM (in its current state) is still a bit daunting for authors of adapters to get their heads around. I'd like to help improve that.

If these docs are okay, I'll start adding more patches here and there, and make some notes from the perspective of helping library authors get up to speed with what they need to do to plug all kinds of query APIs and dataset enumerators into ROM.

Let me know what you think.

### CODE REVIEW

**Comment by aflatter** [comment]
:clap:

**Comment by splattael** [comment]
:+1: :green_heart:

File: lib/rom/lint/linter.rb [hunk 1]
```diff
@@ -1,5 +1,19 @@
 module ROM
   module Lint
+    # Base class for building linters that check source code
+    #
+    # Linters are used by authors of ROM adapters to verify that their
+    # integration comnplies with the ROM api.
```
Comment (muescha) [inline lib/rom/lint/linter.rb]: comnplies -> complies

**Comment by solnic** [comment]
Thanks @maetl for this. It's really nice to hear that you had fun time building an adapter for ROM despite lack of proper documentation :)

**Comment by solnic** [comment]
@maetl I gave you commit access to rom repo in case you'd like to work on some docs or improve some things, thanks for helping out :heart:

File: lib/rom/lint/linter.rb (hunk 1 above)
Comment (maetl) [inline lib/rom/lint/linter.rb]: Thanks for noting. This line was copy/pasted from the original commit message when this file was added.

**Pull request merged**